pydicom/dataelem.py

```diff
@@ -365,7 +365,7 @@ def to_json_dict(
                 else:
                     value = [self.value]
                 json_element['Value'] = [v for v in value]
-        if hasattr(json_element, 'Value'):
+        if 'Value' in json_element:
             json_element['Value'] = jsonrep.convert_to_python_number(
```

Comment:
Indeed, this code made no sense - thanks! Can you add a test that shows the code is now correctly run? E.g. make sure that numerical strings in 'Value' are converted to the correct number type.